lib/docassemble/startserver.js

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env node
+const puppeteer = require('puppeteer');
+const da_i = require('./docassemble_api_interface');
+const session_vars = require('../utils/session_vars');
+
+const fs = require('fs')
+
+const startserver = async () => {
+  // Open the browser, login with env var creds
+  /*
+  let browser = await puppeteer.launch({headless: !session_vars.get_debug()})
+  let page = await browser.newPage();
+  // Go to the sign in page
+  let login_url = `${ session_vars.get_da_server_url() }/user/sign-in`;
+  try {
+    // puppeteer will ensure proper timeout.
+    await page.goto( login_url, { waitUntil: `domcontentloaded`, timeout: 3000 });
+    await page.waitForSelector( `.dabody` );
+  } catch ( error ) {
+    let err_msg = `Error occurred when ALKiln tried to go to "${ login_url }: ${ error }".`
+    console.log(err_msg);
+    throw error;
+  } 
+
+  let email = process.env[ "DA_ADMIN_API_KEY" ];
+  let password = process.env[ "USER_PASSWORD" ];
+
+  // Try to log in
+  await page.type( `#email`, email );
+  await page.type( `#password`, password );
+  let elem = await page.$( `button[name="submit"]` );
+  await elem.click();
+  await page.waitForNavigation({waitUntil: 'domcontentloaded'});
+
+  // Wait to see the success message.
+  // This is a bit of an assumption and needs success and failure testing
+  // TODO: use a Promise.race to fail faster.
+  await page.waitForSelector( `.datopcenter .alert-success` );
+  // make an API key
+
+  // TODO(brycew): with or without the API key?
+  // Install the package in order to also install all of its dependencies
+  let api_key_url = `${ session_vars.get_da_server_url() }/manage_api?action=new`;
+  await page.goto( api_key_url, { waitUntil: `domcontentloaded`, timeout: 3000 });
+  await page.type( `#name`, "ALKiln tmp key");
+  await page.select( `#method`, "none");
+  let winner = await Promise.all([page.click(`#submit`), page.waitForNavigation({waitUntil: 'domcontentloaded'})]);
+  console.log(`submitted`);
+  const api_text = await page.$eval('body p code', elem => elem.innerText);
+  process.env.DOCASSEMBLE_DEVELOPER_API_KEY = api_text;
+  console.log(`api key: ${api_text}`);
+  let file = fs.writeFileSync( ".tmpenv", `export DOCASSEMBLE_DEVELOPER_API_KEY=${api_text}`);
+
+  await page.goto(`${ session_vars.get_da_server_url() }/user/sign-out`, {waitUntil: `domcontentloaded`});  
+  console.log(`signed out`);
+  await browser.close();
+  */
```

Comment:
IDK if any of the above is useful, but I did have a version that could make an API key for the user, but it still needs a user name and password.

Reply:
If we do want to create unique API keys and we do want to keep them secret, we'd need to work with the GitHub API to create that secrets.

Stream of consciousness: Not sure why we need a custom API key - the server gets killed after the tests. Are we concerned people will interfere with the process in the middle of the action? I think only people with permissions could use something like tmate with a test run. It seems like that could be a security measure the authors have to take - only allow trusted folks to be collaborators. Maybe that's too restrictive? If so, does GitHub have a way to have finer-grained permissions - allow only admins to mess with runningtests? Even that might be too restrictive since collaborators do need to look at tests, though I guess they don't need to look at running tests...

Reply:
Whatever API key we make, we shouldn't keep around, so don't need to worry about that.

And I only brought this up in case it might be useful for other Kiln features; it would not be useful for running docassemble on docker on GitHub.

Reply:
It's true it's another way to make an api key. I wish there were a place to store code like this without keeping it in the repo itself. I'll never find it in here. Maybe we can make issues for this kind of stuff with a special label or something.

Reply:
Conclusion: Made https://github.com/SuffolkLITLab/ALKiln/issues/815